external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaTimeUtils.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.apache.spark.sql.kafka010
+
+import MockedSystemTime.currentMockSystemTime
+
+/**
```

Comment:
FYI, there are Clock / ManualClock / SystemClock in `org.apache.spark.util` package. We still need to have a singleton to do the trick anyway, but we can leverage these implementations and only expose the method to advance the manual time.

Reply:
👍